core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -188,6 +221,42 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit {
       });
   }
 
+  onGroupDelete(rows: Set<WorkflowExecutionsEntry>) {
+    const modalRef = this.modalService.open(DeletePromptComponent);
+    modalRef.componentInstance.deletionType = "execution";
+    let deletionNames = "";
+    for (let row of rows) {
+      deletionNames = deletionNames.concat(row.name);
+      deletionNames = deletionNames.concat("; ");
+    }
+    modalRef.componentInstance.deletionName = deletionNames.slice(0, -2);
+    let eIds: number[] = [];
+    for (let row of rows.values()) {
+      eIds.push(row.eId);
+    }
+
+    from(modalRef.result)
+      .pipe(untilDestroyed(this))
+      .subscribe((confirmToDelete: boolean) => {
+        if (confirmToDelete && this.workflow.wid !== undefined) {
+          this.workflowExecutionsService
+            .groupDeleteWorkflowExecutions(this.workflow.wid, eIds)
+            .pipe(untilDestroyed(this))
+            .subscribe({
+              complete: () => {
+                for (let row of rows.values()) {
```

Comment:
why so many for loops?

Reply:
I think this for loop is unavoidable bc when slicing the array, you need to slice one at a time

Reply:
you can use .filter() or any other function to remove all elements that are in some list